lib/cacheable/controller.rb

```diff
@@ -36,14 +36,26 @@ def response_cache(key_data=nil, version_data=nil, &block)
         return yield
       end
 
-      handler = Cacheable::ResponseCacheHandler.new(self) do |h|
-        h.key_data       = key_data       || cache_key_data
-        h.version_data   = version_data   || cache_version_data
-        h.block          = block
-        h.cache_store    = Cacheable.cache_store
+      handler = Cacheable::ResponseCacheHandler.new(
+        key_data:              key_data || cache_key_data,
+        version_data:          version_data || cache_version_data,
+        env:                   request.env,
+        cache_age_tolerance:   cache_age_tolerance_in_seconds,
+        serve_unversioned:     serve_unversioned_cacheable_entry?,
+        force_refill_cache:    force_refill_cache?,
+        headers:               response.headers,
+        &block
+      )
+
+      status, _headers, body = handler.run!
+
+      return if request.env['cacheable.miss']
+      case request.env['cacheable.store']
+      when 'server'
+        render status: status, plain: body.first
+      when 'client'
+        head :not_modified
```

Comment:
From what I understand, this controller would only be called by a Rails app? (because of `#render` and `#head`)

What would happen when we call this from a plain Ruby app? Or am I missing something? 🤔

Reply:
Yeah exactly I made this file only be required if rails is present.

https://github.com/Shopify/cacheable/blob/e33d12db17868fe97fa51de6e59e02599644a87f/lib/cacheable/railtie.rb#L1

It'd throw an error if it wasn't a rails controller.